pkg/chunkenc/memchunk.go

```diff
@@ -1401,13 +1401,12 @@ func newBufferedIterator(ctx context.Context, pool compression.ReaderPool, b []b
 	stats := stats.FromContext(ctx)
 	stats.AddCompressedBytes(int64(len(b)))
 	return &bufferedIterator{
-		stats:                  stats,
-		origBytes:              b,
-		reader:                 nil, // will be initialized later
-		pool:                   pool,
-		format:                 format,
-		symbolizer:             symbolizer,
-		currStructuredMetadata: structuredMetadataPool.Get().(labels.Labels),
```

Comment:
I removed the structuredMetadataPool because it's no longer possible for `labels.Labels` to be constructed using shared memory. 

The only way to do it now is with `labels.ScratchBuilder.Overwrite`, which shares memory between the scratch builder and the labels instance. If we notice allocation problems, we can explore updating this to use Overwrite.